File: admin/src/api/webmail.ts
import { http } from '@/utils/http'

export interface ApiResp<T> {
  code: number
  message: string
  data: T
}

export interface LqqqMailItem {
  subject: string
  date: string
  mailbox: string
  code: string
  view_href: string
  html_body?: string
  body?: string
}

export interface LqqqMailDetail {
  body: string
  html_body: string
  code: string
}

export interface WebmailMailDetail {
  body: string
  html_body: string
  code: string
}

export interface LqqqFetchData {
  email: string
  inbox: LqqqMailItem[]
  junk: LqqqMailItem[]
}

export interface ToolsvipMailItem {
  subject: string
  date: string
  mailbox: string
  code: string
  html_body: string
  body: string
}

export interface ToolsvipFetchData {
  email: string
  inbox: ToolsvipMailItem[]
  junk: ToolsvipMailItem[]
}

export const fetchLqqqMails = (account_line: string, account_format = '1') =>
  http.post<ApiResp<LqqqFetchData>>('/admin/webmail/lqqq/fetch', { account_line, account_format })

export const fetchLqqqDetail = (
  account_line: string,
  view_href: string,
  account_format = '1',
) =>
  http.post<ApiResp<LqqqMailDetail>>('/admin/webmail/lqqq/detail', {
    account_line,
    view_href,
    account_format,
  })

export const fetchToolsvipMails = (account_line: string, account_format = '1') =>
  http.post<ApiResp<ToolsvipFetchData>>('/admin/webmail/toolsvip/fetch', { account_line, account_format })

File: backend/controller/webmail.go
package controller

import (
	"net/http"

	"github.com/gin-gonic/gin"
	"github.com/kingfer30/topup-online/utils/webmail"
)

type webmailFetchRequest struct {
	AccountLine   string `json:"account_line" binding:"required"`
	AccountFormat string `json:"account_format"`
}

// LqqqFetch 获取 lqqq 邮件列表
func LqqqFetch(c *gin.Context) {
	var req webmailFetchRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"code": 400, "message": "参数错误: " + err.Error()})
		return
	}

	line := firstNonEmptyLine(req.AccountLine)
	email, password, err := webmail.ParseWebMailAccountLine(line, req.AccountFormat)
	if err != nil {
		c.JSON(http.StatusOK, gin.H{"code": 400, "message": err.Error()})
		return
	}

	inbox, junk, err := webmail.FetchLqqqMails(email, password)
	if err != nil {
		c.JSON(http.StatusOK, gin.H{"code": 500, "message": "取件失败: " + err.Error()})
		return
	}

	c.JSON(http.StatusOK, gin.H{
		"code":    200,
		"message": "成功",
		"data": gin.H{
			"email": email,
			"inbox": inbox,
			"junk":  junk,
		},
	})
}

type lqqqDetailRequest struct {
	AccountLine   string `json:"account_line" binding:"required"`
	AccountFormat string `json:"account_format"`
	ViewHref      string `json:"view_href" binding:"required"`
}

// LqqqDetail 拉取 lqqq 单封邮件详情
func LqqqDetail(c *gin.Context) {
	var req lqqqDetailRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"code": 400, "message": "参数错误: " + err.Error()})
		return
	}

	line := firstNonEmptyLine(req.AccountLine)
	email, password, err := webmail.ParseWebMailAccountLine(line, req.AccountFormat)
	if err != nil {
		c.JSON(http.StatusOK, gin.H{"code": 400, "message": err.Error()})
		return
	}

	detail, err := webmail.FetchLqqqMailDetail(email, password, req.ViewHref)
	if err != nil {
		c.JSON(http.StatusOK, gin.H{"code": 500, "message": err.Error()})
		return
	}

	c.JSON(http.StatusOK, gin.H{
		"code":    200,
		"message": "成功",
		"data":    detail,
	})
}

// ToolsvipFetch 获取 toolsvip 邮件列表
func ToolsvipFetch(c *gin.Context) {
	var req webmailFetchRequest
	if err := c.ShouldBindJSON(&req); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"code": 400, "message": "参数错误: " + err.Error()})
		return
	}

	line := firstNonEmptyLine(req.AccountLine)
	email, password, err := webmail.ParseWebMailAccountLine(line, req.AccountFormat)
	if err != nil {
		c.JSON(http.StatusOK, gin.H{"code": 400, "message": err.Error()})
		return
	}

	inbox, junk, err := webmail.FetchToolsvipMails(email, password)
	if err != nil {
		c.JSON(http.StatusOK, gin.H{"code": 500, "message": "取件失败: " + err.Error()})
		return
	}

	c.JSON(http.StatusOK, gin.H{
		"code":    200,
		"message": "成功",
		"data": gin.H{
			"email": email,
			"inbox": inbox,
			"junk":  junk,
		},
	})
}
